nomos-da/network/messages/src/dispersal.rs

```diff
@@ -33,25 +38,47 @@ impl DispersalError {
 }
 
 #[repr(C)]
-#[derive(Debug, Eq, PartialEq, Serialize, Deserialize)]
-pub struct DispersalRequest {
-    pub share: Share,
-    pub subnetwork_id: SubnetworkId,
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub enum DispersalRequest {
+    Share(ShareRequest),
+    Tx(SignedMantleTx),
 }
 
 impl DispersalRequest {
     #[must_use]
-    pub const fn new(share: Share, subnetwork_id: SubnetworkId) -> Self {
-        Self {
-            share,
-            subnetwork_id,
+    pub fn blob_id(&self) -> Option<BlobId> {
+        match self {
+            Self::Share(share) => Some(share.share.blob_id),
+            Self::Tx(tx) => match tx.mantle_tx.ops.first() {
+                Some(Op::ChannelBlob(blob_op)) => Some(blob_op.blob),
+                _ => None,
+            },
```

Comment:
This should probably be a convenience method in the SignedMantleTx or something like that.

Reply:
> This should probably be a convenience method in the SignedMantleTx or something like that.

Will work on this in upcoming ledger related tasks, this will be part of op validation.